cli/tests/test_undo_redo_commands.rs

```diff
@@ -137,6 +137,24 @@ fn test_undo_with_rev_arg_falls_back_to_revert() {
     │  args: jj undo @-
     [EOF]
     ");
+
+    let output = work_dir.run_jj(["undo", "--what", "repo"]);
```

Comment:
`undo --what` is deprecated and will be removed. If there are tests for `op restore --what`, I don't think we need the test for `undo`.